### ISSUE TITLE
Unable to do a POST using a URL image

### ISSUE THREAD

**Issue by wiazur** [tomas/needle#317]
I keep getting an error saying the URL is invalid. The URL works fine, it has an image on the other end. I'm thinking I have my code constructed incorrectly, or maybe there is a bug? Nothing about URL images in the documentation and no one on Stack Overflow knew what to do. The API I am calling has been confirmed to work with a different sample.

```
const imageUrl = 'https://upload.wikimedia.org/wikipedia/commons/3/37/Dagestani_man_and_woman.jpg'

// Request parameters.
const params = {
    returnFaceId: true,
    returnFaceLandmarks: false,
    returnFaceAttributes: 'age,gender,headPose,smile,facialHair,glasses,emotion,hair,makeup,occlusion,accessories,blur,exposure,noise'
}

var options = {
    body: '{"url": ' + '"' + imageUrl + '"}',
    headers: {
        'Content-Type': 'application/json',
        'Ocp-Apim-Subscription-Key': subscriptionKey
    }
}

needle.post(endpoint, params, options, function (err, res, body) {
    console.log(`Status: ${res.statusCode}`)
    console.log('Body: ', body)
    console.log('ERROR: ' + err)
    //console.log(res)
})
```

Output:
```
Status: 400
Body:  { error: { code: 'InvalidURL', message: 'Invalid image URL.' } }
ERROR: null
```

**Comment by JiPaix** [comment]
I believe this is Face API (which i'm not familiar with), but according to this [documentation](https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395250) you are receiving a response : 
`{ error: { code: 'InvalidURL', message: 'Invalid image URL.' } }`

It means needle did post `'{"url": ' + '"' + imageUrl + '"}'` and Face API recognized the `url` key with a string value and your endpoint is processing the string incorrectly.

**Comment by wiazur** [comment]
Thanks @JiPaix, Ok, just wanted to check. I did read that in the API, but just not sure _how_ it's processing it incorrectly. I've put in a query with the source SDK as well, so maybe something will come out of that. But thanks!

**Comment by JiPaix** [comment]
Have you tried querying their API directly from the doc page? did you get the same error ?
- Select endpoint
![image](https://user-images.githubusercontent.com/26584973/82953936-fc56ea80-9fab-11ea-9f92-66df5fbd82d2.png)
- manually fill the fields and click send.
![image](https://user-images.githubusercontent.com/26584973/82954084-48099400-9fac-11ea-889d-af88591b5ade.png)

**Comment by wiazur** [comment]
Thanks @JiPaix, I did actually, it tested out fine. So trying to figure out why these other libraries aren't working. I did not mention, but I tried the Axios library too and got a similar result. But the old Node requests library works OK. So it's a real mystery. The requests library is deprecated though, so I need to change it. It's possible it's a bug with the Face service. Thanks for helping me troubleshoot though!
